UM-USElab/gradecraft-development
# Unlocks beneath

### Status
**READY**

### Description
This PR makes it possible to create unlock conditions for the Assignment Grade Earned condition, where student unlock new opportunities based on having earned less than a value or between a range of values on a particular assignment. 

### Todos
- [x] Consider whether to add student-facing clarification of unlock above vs. below (and point value threshold)
- [x] Fix minimum and maximum threshold bug.

### Migrations
YES

### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.
1. As an instructor, create a new Unlock Condition specifying an assignment, and selecting the condition "Grade Earned". 
2. Specify a maximum value. 
3. Go to the qualifying assignment and assess a student to have higher than that value - confirm that they have _not_ unlocked the associated assignment. 

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Unlock conditions
* Assignment show page

======================
